Wallet Decrypt Redesign

wbobeirne opened the pull request:
Closes #565 and addresses a few of #557's checkboxes. Rather than list all the changes I made, I think it's just best to watch it in action.

## Screenshots / Gfycats

### [Desktop Account Unlock (Gfycat)](https://gfycat.com/SpicySoulfulEagle)
### [Mobile Account Unlock (Gfycat)](https://gfycat.com/VelvetySpanishAmericantoad)

#### Sign & Verify
<img width="1177" alt="screen shot 2017-12-28 at 12 49 58 pm" src="https://user-images.githubusercontent.com/649992/34419065-b73b44ec-ebcf-11e7-8c1e-81a9ebb37e3b.png">

#### Contract Deploy
<img width="1162" alt="screen shot 2017-12-28 at 12 56 30 pm" src="https://user-images.githubusercontent.com/649992/34419073-c51317ca-ebcf-11e7-9ada-a902794ea6ef.png">

#### Contract Interact
<img width="1177" alt="screen shot 2017-12-28 at 12 56 52 pm" src="https://user-images.githubusercontent.com/649992/34419086-db7ed2d8-ebcf-11e7-9040-bec447835b2e.png">


## Caveats

* Some people might not be into the animations. Let me know what you think of them.
* Expanding to additional wallets (Keepkey?) will be a little more difficult to fit into this design than adding another radio button.
* WalletDecrypt.tsx's code has become a little ugly for my tastes. Would love feedback on that.
* I think there are a few outstanding PRs that touch unlock. I'll be sure to handle their conflicts, whoever's get merged first.
* The wrapping it does in some at some of the intermediate sizes is a little janky. Ideally they'd wrap 2 by 2, but the markup required for that was pretty dirty.

<img width="280" alt="screen shot 2017-12-28 at 1 02 54 pm" src="https://user-images.githubusercontent.com/649992/34419036-919f6588-ebcf-11e7-910b-46affe82cb64.png">

wbobeirne commented on common/containers/Tabs/SignAndVerifyMessage/components/SignMessage/index.tsx:
```diff
@@ -44,42 +44,43 @@ export class SignMessage extends Component<Props, State> {
+            )}
+          </div>
+        ) : (
+          <WalletDecrypt hidden={unlocked} disabledWallets={DISABLE_WALLETS.UNABLE_TO_SIGN} />
```
@skubakdj this was my suggested change for how we use the disabledWallets prop. LMK if it looks good to you.

wbobeirne commented on common/sass/mixins.scss:
```diff
@@ -69,3 +69,30 @@
+    transform: translate(-50%, -100%) translateY(-8px);
+    transition-delay: 400ms, 400ms, 300ms;
+  }
+}
```
I felt that this was the lowest friction way to implement tooltips, instead of making the component take in a whole bunch of props for showing and hiding. You simply add this mixin to the parent element you want to show the tooltip on hover. Let me know if y'all would rather have it be component props. Maybe we can do either / or?

wbobeirne commented on common/translations/lang/el.json:
```diff
@@ -82,7 +82,7 @@
-    "x_Keystore2": "Αρχείο Keystore (UTC / JSON) ",
+    "x_Keystore2": "Αρχείο Keystore ",
     "x_KeystoreDesc": "Αυτό το αρχείο Keystore έχει την ίδια μορφή που χρησιμοποιείται από το Mist ώστε να μπορείτε εύκολα να το εισάγετε στο μέλλον. Είναι το συνιστώμενο αρχείο για λήψη και δημιουργία αντιγράφου ασφαλείας. ",
     "x_MetaMask": "Metamask / Mist ",
```
TODO: Do the same Metamask / Mist split I did in en.json to all languages.

dternyak commented:
Still need to review implementation, but great work from a usability perspective 👍 

A few issues I found on the UI / usability side:

1. Inability to change wallets after decrypting: 
     - sign message
     - deploy contract

2. Padding(?) issues with helper text
<img width="350" alt="screen shot 2017-12-28 at 12 25 16 pm" src="https://user-images.githubusercontent.com/7861465/34419661-b8e63072-ebca-11e7-81bd-37647bd68642.png">

3. Significant white space / relatively small input fields. In the case of view-only wallets, users will need to scroll horizontally to see the entire pasted address, but there is plenty of whitespace to expand into.

<img width="1273" alt="screen shot 2017-12-28 at 12 26 00 pm" src="https://user-images.githubusercontent.com/7861465/34419734-2367f89a-ebcb-11e7-8e7b-c6c64950aab9.png">

(Opinionated) I think vertically and horizontally centering the unlock view would be an improvement.
<img width="1323" alt="screen shot 2017-12-28 at 12 27 13 pm" src="https://user-images.githubusercontent.com/7861465/34419744-3821e44e-ebcb-11e7-9974-8d116f9c3cc9.png">

Will be back with a code review later today 😄

wbobeirne commented on common/translations/lang/en.json:
```diff
@@ -81,9 +81,11 @@
     "x_JsonDesc": "This is the unencrypted, JSON format of your private key. This means you do not need the password but anyone who finds your JSON can access your wallet & Ether without the password. ",
     "x_Keystore": "Keystore File (UTC / JSON · Recommended · Encrypted) ",
-    "x_Keystore2": "Keystore File (UTC / JSON) ",
+    "x_Keystore2": "Keystore File ",
```
Is this change OK? Or should I make `x_Keystore3`?

dternyak replied:
@tayvano Feel free to chime in with your thoughts.

My understanding is we'll do a once-over for all of the copy, and can address issues at that point.

dternyak commented:
While MetaMask is likely the most popular option, we may want to make users aware than any web3 provider is supported (for the first option).

wbobeirne commented:
> While MetaMask is likely the most popular option, we may want to make users aware than any web3 provider is supported.

I forgot to mention this up top, but it detects metamask and shows that. Also detects mist:

### With no known web3 client detected

<img width="1277" alt="screen shot 2017-12-30 at 8 41 22 pm" src="https://user-images.githubusercontent.com/649992/34458471-3678821e-eda2-11e7-80db-d9388c8778fc.png">


### With Mist

![screen shot 2017-12-30 at 8 42 34 pm](https://user-images.githubusercontent.com/649992/34458472-3a08cf42-eda2-11e7-9109-eb0ce5f1ddd1.png)

wbobeirne commented:
@dternyak 

1. Added "Change Wallet" buttons on sign and deploy tabs.
2. Removed outline on these buttons. They have hover and active states, so it should be fine without.
3. Height is now auto instead of being as big for unlock, so the whitespaciness should feel better.
